catalog/async/catalog-async-processingframework/src/main/java/org/codice/ddf/catalog/async/processingframework/impl/InMemoryProcessingFramework.java

```diff
@@ -0,0 +1,257 @@
+                                plugin.getClass()
+                                        .getCanonicalName(),
+                                e);
+                    }
+                }
+
+                storeProcessRequest(request);
+            });
+        }
+    }
+
+    @Override
+    public void submitUpdate(ProcessRequest<ProcessUpdateItem> input) {
+        if (postProcessPlugins == null || postProcessPlugins.isEmpty()) {
+            LOGGER.debug("postProcessPlugins is empty. Not starting post process thread");
+        } else {
+            threadPool.submit(() -> {
+                ProcessRequest<ProcessUpdateItem> request = input;
+
+                for (PostProcessPlugin plugin : postProcessPlugins) {
+                    try {
+                        request = plugin.processUpdate(request);
+                    } catch (PluginExecutionException e) {
+                        LOGGER.debug("Unable to process update request through plugin: {}",
+                                plugin.getClass()
+                                        .getCanonicalName(),
+                                e);
+                    }
+                }
+
+                storeProcessRequest(request);
+            });
+        }
+    }
+
+    @Override
+    public void submitDelete(ProcessRequest<ProcessDeleteItem> input) {
+        if (postProcessPlugins == null || postProcessPlugins.isEmpty()) {
+            LOGGER.debug("postProcessPlugins is empty. Not starting post process thread");
+        } else {
+            threadPool.submit(() -> {
+                ProcessRequest<ProcessDeleteItem> request = input;
+
+                for (PostProcessPlugin plugin : postProcessPlugins) {
+                    try {
+                        request = plugin.processDelete(request);
+                    } catch (PluginExecutionException e) {
+                        LOGGER.debug("Unable to process request through plugin: {}",
+                                plugin.getClass()
+                                        .getCanonicalName(),
+                                e);
+                    }
+                }
+            });
+        }
+    }
+
+    private <T extends ProcessResourceItem> void storeProcessRequest(
+            ProcessRequest<T> processRequest) {
+        LOGGER.trace("Storing update request post processing change(s)");
+
+        Set<ContentItem> contentItemsToUpdate = new HashSet<>();
+        Set<Metacard> metacardsToUpdate = new HashSet<>();
+
+        for (T item : processRequest.getProcessItems()) {
+            if (item.isMetacardModified()) {
+                metacardsToUpdate.add(item.getMetacard());
+            }
+
+            final ProcessResource processResource = item.getProcessResource();
+            if (processResource != null && processResource.isModified()) {
+                try {
+                    byte[] byteArray = IOUtils.toByteArray(processResource.getInputStream());
```

Comment:
The more I think about this, we should not allow a `ProcessResource` to be made without an `InputStream` as a `ProcessResource` wouldn't be very useful without it since that is the item it is supposed to be representing. I'm going to remove the `@Nullable` from the `ProcessResouce#getInputStream()` method.